src/EurekaClient.js

```diff
@@ -51,6 +51,10 @@ export default class Eureka extends EventEmitter {
 
     this.logger.debug('initializing eureka client');
 
+    if (typeof config.requestMiddleware === 'function') {
```

Comment:
Just a suggestion, but may make the code a bit easier to read if you default this to a no-op middleware that just returns the provided request config. i.e.:

```
this.requestMiddleware = config.requestMiddleware || (req => req);
```